`control_statement` asks to illegally unwrap guard.

### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`control_statement` checks to ensure the control statement is not unnecessarily wrapped in parenthesis. However, parenthesis are required in certain cases for control statements, such as when the statement itself consists of braces, to disambiguate the braces from the control operator's braces. The following example illustrates where `control_statement` demands the removal of parentheses which would render the statement illegal in the language.

```swift
guard ({ switch target {
    case a: true
    case b: false
}
}())
else { return false }
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/Users/hubstaff/Developer/hubstaff/ios/test.swift:1:1: warning: Control Statement Violation: `if`, `for`, `guard`, `switch`, `while`, and `catch` statements shouldn't unnecessarily wrap their conditionals or arguments in parentheses (control_statement)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure): 0.57.0
* Xcode version (run `xcodebuild -version` to be sure): 16.1, 16B40
* Installation method used (Homebrew, CocoaPods, building from source, etc): brew

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,6 +40,11 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#5787](https://github.com/realm/SwiftLint/issues/5787)
 
+* Stop triggering the `control_statement` rule on closures being directly
+  called as conditions.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#5846](https://github.com/realm/SwiftLint/issues/5846)
+
 * Do not trigger `self_in_property_initialization` rule on `self` in
   key paths expressions.  
   [SimplyDanny](https://github.com/SimplyDanny)

--- a/Source/SwiftLintBuiltInRules/Rules/Style/ControlStatementRule.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Style/ControlStatementRule.swift
@@ -29,6 +29,8 @@ struct ControlStatementRule: Rule {
             Example("switch (lhs, rhs) {}"),
             Example("if (f() { g() {} }) {}"),
             Example("if (a + f() {} == 1) {}"),
+            Example("if ({ true }()) {}"),
+            Example("if ({if i < 1 { true } else { false }}()) {}", excludeFromDocumentation: true),
         ],
         triggeringExamples: [
             Example("↓if (condition) {}"),
@@ -176,7 +178,7 @@ private extension ExprSyntax {
    private func containsTrailingClosure(_ node: Syntax) -> Bool {
         switch node.as(SyntaxEnum.self) {
         case .functionCallExpr(let node):
-            node.trailingClosure != nil
+            node.trailingClosure != nil || node.calledExpression.is(ClosureExprSyntax.self)
         case .sequenceExpr(let node):
             node.elements.contains { containsTrailingClosure(Syntax($0)) }
         default: false

